mdm/api/src/main/java/com/gentics/mesh/core/data/dao/SchemaDao.java

```diff
@@ -330,135 +116,31 @@ default void addSchema(HibSchema schemaContainer, HibProject project, HibUser us
-		Throwable error = indexHandler.validate(schema).blockingGet(10, TimeUnit.SECONDS);
-
-		if (error != null) {
-			if (error instanceof GenericRestException) {
-				throw (GenericRestException) error;
-			} else {
-				throw new RuntimeException(error);
-			}
-		}
-	}
-
-	@Override
-	default HibSchema create(HibProject root, InternalActionContext ac, EventQueueBatch batch, String uuid) {
-		HibSchema schema = create(ac, batch, uuid);
-		addSchema(schema, root, ac.getUser(), batch);
-		Tx.get().persist(root, Tx.get().projectDao());
-		assignEvents(schema, UNASSIGNED).forEach(batch::add);
-		return schema;
-	}
-
-	@Override
-	default SchemaResponse transformToRestSync(HibSchema element, InternalActionContext ac, int level,
-			String... languageTags) {
-		return element.transformToRestSync(ac, level, languageTags);
-	}
-
-	@Override
-	default boolean update(HibSchema element, InternalActionContext ac, EventQueueBatch batch) {
-		throw new NotImplementedException("Updating is not directly supported for schemas. Please start a schema migration");
-	}
-
-	@Override
-	default boolean update(HibProject project, HibSchema element, InternalActionContext ac, EventQueueBatch batch) {
-		return update(element, ac, batch);
-	}
-
-	@Override
-	default void unlink(HibSchema schema, HibProject project, EventQueueBatch batch) {
-		removeSchema(schema, project, batch);
-	}
-
-	@Override
-	default boolean isLinkedToProject(HibSchema schema, HibProject project) {
-		return contains(project, schema);
-	}
-
-	@Override
-	default FieldSchemaContainerComparator<SchemaModel> getFieldSchemaContainerComparator() {
-		return Tx.get().data().schemaComparator();
-	}
-
-	@Override
-	default void deleteVersion(HibSchemaVersion version, BulkActionContext bac) {
-		generateUnassignEvents(version).forEach(bac::add);
-		// Delete change
-		HibSchemaChange<?> change = version.getNextChange();
-		if (change != null) {
-			deleteChange(change, bac);
-		}
-		// Delete referenced jobs
-		for (HibJob job : version.referencedJobsViaFrom()) {
-			job.remove();
-		}
-		for (HibJob job : version.referencedJobsViaTo()) {
-			job.remove();
-		}
-		// Delete version
-		Tx.get().delete(version, version.getClass());
-	}
+	Result<HibProject> findLinkedProjects(HibSchema schema);
 
 	/**
-	 * Genereates branch unassign events for every assigned branch.
+	 * Load all nodes.
 	 * 
+	 * @param version
+	 * @param uuid
+	 * @param user
+	 * @param type
 	 * @return
 	 */
-	private Stream<BranchSchemaAssignEventModel> generateUnassignEvents(HibSchemaVersion version) {
-		return getBranches(version).stream()
-			.map(branch -> branch.onSchemaAssignEvent(version, UNASSIGNED, null));
-	}
+	Result<? extends HibNode> findNodes(HibSchemaVersion version, String uuid, HibUser user, ContainerType type);
+
+	/**
+	 * Add the schema to the db.
+	 * 
+	 * @param schema
+	 */
+	void addSchema(HibSchema schema);
```

Comment:
Isn't this a low-level method and should be moved to PersistingSchemaDao?
Outside of PersistingSchemaDao, it is used by tests only.

Reply:
This method violates the DAO API strategy, making the duplicated things of the LLDB, and has been removed.